Deprecate username:password login in favor of access_key:access_id

jazzdan opened the pull request:
Add note about how authenticating with username and password is deprecated.
Explain how to generate an access key/ID and how to use that in the `sumo_creds` file.

Commit:
Update README.md

nahiluhmot commented:
Seems good. Do we want to change the keys to `access_key` and `id` to reflect that?

jazzdan commented:
@nahiluhmot I think that'd be a good idea! I can work that in to this PR.

Commit:
Deprecate email/password login

This commit does two things:

1) It adds a new exception (`Sumo::Error::EmailPasswordDeprecated`)
that is thrown when a credential file is supplied that contains the
deprecated fields `email` or `password`.
2) It removes all mention of passwords and emails, replacing the terms
with `access_key` and `access_id` respectively.

jazzdan commented:
@nahiluhmot I've reworked the PR in 96e3a82 to completely deprecate email/password login. Let me know what you think!

nahiluhmot commented:
@jazzdan Merge when ready -- looks great!

jazzdan commented:
Thanks @nahiluhmot!
